commit 853faa22949f34133ec49f7fa2ad015dbaae5b9f
Author: Zbigniew JÄ™drzejewski-Szmek <zbyszek@in.waw.pl>
Date:   Thu Aug 18 21:39:39 2016 -0400

    yummy:patchX:Revert--pid1--reconnect-to-the-console-before-being-re-executed-.patch

Revert "pid1: reconnect to the console before being re-executed"

    
    This reverts commit affd7ed1a923b0df8479cff1bd9eafb625fdaa66.
    
    > So it looks like make_console_stdio() has bad side effect. More specifically it
    > does a TIOCSCTTY ioctl (via acquire_terminal()) which sees to disturb the
    > process which was using/owning the console.
    
    Fixes #3842.
    https://bugs.debian.org/834367
    https://bugzilla.redhat.com/show_bug.cgi?id=1367766
    
    (cherry picked from commit bd64d82c1c0e3fe2a5f9b3dd9132d62834f50b2d)

diff --git a/src/core/main.c b/src/core/main.c
index f2adca7..f59a55f 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -2016,9 +2016,6 @@ finish:
                                 log_error_errno(r, "Failed to switch root, trying to continue: %m");
                 }
 
-                /* Reopen the console */
-                (void) make_console_stdio();
-
                 args_size = MAX(6, argc+1);
                 args = newa(const char*, args_size);
 
@@ -2066,6 +2063,9 @@ finish:
                 arg_serialization = safe_fclose(arg_serialization);
                 fds = fdset_free(fds);
 
+                /* Reopen the console */
+                (void) make_console_stdio();
+
                 for (j = 1, i = 1; j < (unsigned) argc; j++)
                         args[i++] = argv[j];
                 args[i++] = NULL;
